crates/nu-cli/src/cli.rs

```diff
@@ -625,7 +639,7 @@ pub fn set_rustyline_configuration() -> (Editor<Helper>, IndexMap<String, Value>
                         rl.set_completion_type(completion_type);
                     }
                     "completion_prompt_limit" => {
-                        let completion_prompt_limit = value.as_u64().unwrap_or(1) as usize;
+                        let completion_prompt_limit = value.as_u64().unwrap_or(100) as usize;
```

Comment:
What do you think of this:

```rust
if let Some(completion_prompt_limit) = value.as_u64() {
  rl.set_completion_prompt_limit(completion_prompt_limit as usize);
}
```

Same for all of the other lines in here where we're duplicating one of the above default values. This means we only change the config if we actually have a value.

Reply:
Apparently it's a bad idea because I changed them all and it won't compile.
```
error[E0308]: mismatched types
   --> crates\nu-cli\src\cli.rs:602:32
    |
602 |                         if let Some(max_history_size) = value.as_u64() {
    |                                ^^^^^^^^^^^^^^^^^^^^^^   -------------- this expression has type `std::result::Result<u64, nu_errors::ShellError>`
    |                                |
    |                                expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected enum `std::result::Result<u64, nu_errors::ShellError>`
               found enum `std::option::Option<_>````

Reply:
In that case you can use `if let Ok(max_history_size) ...`, because it's a `Result`, not an `Option` 🙂

Reply:
I assumed an `Option`, btw, but these could all be `Result`s

Reply:
ok, i'll give that a whirl. Thx.